R/match_pc_gpprac.R

```diff
@@ -0,0 +1,379 @@
+      tidyr::fill(postcode, .direction = "up") %>%
+      dplyr::ungroup()
+  } else if (type == "gp") {
+    return_data <- dplyr::left_join(data_gpprac_match, data_for_matching, by = "chi") %>%
+      dplyr::filter(!is_missing(chi) & !is.integer(all_match)) %>%
+      dplyr::group_by(chi) %>%
+      # Arrange by one of the keydates so the most recent gpprac is at the top even if it's NA
+      # Then fill the values of gpprac upwards so the NA is filled in
+      dplyr::arrange(
+        dplyr::desc(is.na(gpprac)),
+        dplyr::desc(keydate2_dateformat),
+        .by_group = TRUE
+      ) %>%
+      dplyr::mutate(gpprac = dplyr::if_else(gpprac == 99999L, NA_real_, gpprac)) %>%
+      tidyr::fill(postcode, .direction = "up") %>%
+      dplyr::ungroup()
+  }
+
+  return(return_data)
+}
+
+#' Use the postcode from the most recent contact with an
+#' individual to fill missing data
+#'
+#' @param data A data frame
+#' @type Either pc for postcodes or gp for GP practices
+#'
+#' @return A data frame with some missing data filled
+<<<<<<< HEAD
+fill_data_from_keydate <- function(data, type = c("pc", "gp")) {
+=======
+fill_data_from_keydate <- function(data, var_to_fill, type = c("pc", "gp")) {
+>>>>>>> 3e001fde84178abc0abe8e038bb6952e69a1111c
+  if (type == "pc") {
+    na_type <- NA_character_
+    return_data <- data %>%
+      dplyr::group_by(chi) %>%
+      # Determine whether a chi has multiple postcodes
+      dplyr::mutate(count = dplyr::n_distinct(postcode)) %>%
+      dplyr::select("chi", "postcode", "count", "keydate2_dateformat") %>%
+      # Put the most recent contact at the top of each chi group
+      dplyr::arrange(dplyr::desc(keydate2_dateformat), .by_group = TRUE) %>%
+      # Recode any postcodes that are not the most recent with NA
+      dplyr::mutate(postcode = dplyr::if_else(dplyr::row_number() != 1,
+                                                 na_type,
+                                                 postcode
+      )) %>%
+      # Fill all the new NAs with the most recent postcode
+      tidyr::fill(postcode, .direction = "down")
+  } else if (type == "gp") {
+    na_type <- NA_integer_
+    return_data <- data %>%
+      dplyr::group_by(chi) %>%
+      # Determine whether a chi has multiple postcodes
+      dplyr::mutate(count = dplyr::n_distinct(gpprac)) %>%
+      dplyr::select("chi", "gpprac", "count", "keydate2_dateformat") %>%
+      # Put the most recent contact at the top of each chi group
+      dplyr::arrange(dplyr::desc(keydate2_dateformat), .by_group = TRUE) %>%
+      # Recode any postcodes that are not the most recent with NA
+      dplyr::mutate(var_to_fill = dplyr::if_else(dplyr::row_number() != 1,
+                                                 na_type,
+                                                 gpprac
+      )) %>%
+      # Fill all the new NAs with the most recent postcode
+      tidyr::fill(gpprac, .direction = "down")
+  }
+
+  return(return_data)
+}
```

Comment:
I think we could try trim this down with something similar to my above comment as this is repeating code for both pc and gpprac variables